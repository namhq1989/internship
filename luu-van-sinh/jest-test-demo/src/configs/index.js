import AppConst from './app';
import MessageConst from './message'
import ImageConst from './image'
import key from './key'
import URLConst from './url'

export {
  AppConst,
  MessageConst,
  ImageConst,
  key,
  URLConst,
}
