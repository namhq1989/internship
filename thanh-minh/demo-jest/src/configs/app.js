export default {

  endpoint: 'https://dev-api.zody.vn',
  // Screen size
  screens: {
    'xs-max': 480,
    'sm-min': 481,
    'sm-max': 767,
    'md-min': 768,
    'md-max': 991,
    'lg-min': 992,
    'lg-max': 1199,
    'xl-min': 1200
  },
  // Notification level
  notification: {
    success: 'success',
    error: 'error',
    warning: 'warning',
    info: 'info'
  },

  // Regex
  regex: {
    email: /\S+@\S+\.\S+/,
    strongPassword: /^(?=.*[A-Z].*[A-Z])(?=.*[!@#$&*])(?=.*[0-9].*[0-9])(?=.*[a-z].*[a-z].*[a-z]).{8}$/,
    username: /^[\w\d]{3,}$/,
    ageInt: /^-{0,1}\d+$/
  },

  // localStorage
  localStorage: {
    accounts: 'accounts',
    accountLogin: 'accountLogin'
  },
}
