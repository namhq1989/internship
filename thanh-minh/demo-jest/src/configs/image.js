export default {
  imageLogin: require('../assets/images/img-zody-login.png'),
  imageIconEngland: require('../assets/images/icon-england.png'),
  imageIconVietNamese: require('../assets/images/vn-icon.png'),
}
