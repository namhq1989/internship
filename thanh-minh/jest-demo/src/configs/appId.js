
export default {
  appId: '195477857985857',
  version: 'v1.0',
  csrf: 'Demo-Account-Kit',
  language: 'vi_VN',
  loginType: 'PHONE'
}
