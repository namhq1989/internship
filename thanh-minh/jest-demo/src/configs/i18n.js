import i18n from 'i18next'
import LanguageDetector from 'i18next-browser-languagedetector'
import MessageConst from './message'

i18n.use(LanguageDetector).init({
  // we init with resources
  resources: {
    en: {
      translations: {
	'Account Kit': 'Login by phone',
        'language-o': 'en_US',
        'login-button': 'Login',
        'email-palceholder': 'Enter email',
        'password-placeholder': 'Enter password',
        'language- label': 'Language',
        'Do not have an account': 'Do not have an account?',
        'Sign-up': 'Sign Up',
        'Username-placeholder': 'Enter username',
        'repassword-placeholder': 'Enter confirm password',
        'age-placeholder': 'Enter age',
        'Do have an account': 'Do you have an account?',
        'log-out': 'Log Out',
        'welcome-page': 'Welcome',
        'come-page': ' to HomePage',
        // validator register
        RequireRegister: MessageConst.en.Register.RequireRegister,
        EmailIsNotValid: MessageConst.en.Register.EmailIsNotValid,
        userNameLengthBetweenFrom2To128Characters: MessageConst.en.Register.userNameLengthBetweenFrom2To128Characters,
        PasswordLengthMustAtLeast6Characters: MessageConst.en.Register.PasswordLengthMustAtLeast6Characters,
        Successfully: MessageConst.en.Register.Successfully,
        ComfirmToFirstPassword: MessageConst.en.Register.ComfirmToFirstPassword,
        AgeMinMax: MessageConst.en.Register.AgeMinMax,
        AgeIntegerNumber: MessageConst.en.Register.AgeIntegerNumber,
        EmailExists: MessageConst.en.Register.EmailExists,
        SuccessRegister: MessageConst.en.Register.SuccessRegister,

        // validator login
        RequireEmailAndPassword: MessageConst.en.Login.RequireEmailAndPassword,
        ErrorAccount: MessageConst.en.Login.ErrorAccount,
        LoginSuccess: MessageConst.en.Login.LoginSuccess,
        NotExistAccounts: MessageConst.en.Login.NotExistAccounts
      }
    },
    vi: {
      translations: {
	'Account Kit': 'Đăng nhập bằng số  điện thoại',
        'language-o': 'vi_VN',
        'login-button': 'Đăng nhập',
        'email-palceholder': 'Nhập email',
        'language- label': 'Ngôn ngữ',
        'Do not have an account': 'Bạn chưa có tài khoản?',
        'Sign-up': 'Đăng ký',
        'Username-placeholder': 'Nhập username',
        'password-placeholder': 'Nhập mật khẩu',
        'repassword-placeholder': 'Nhập xác nhận mật khẩu',
        'age-placeholder': 'Nhập tuổi',
        'Do have an account': 'Bạn có tài khoản?',
        'log-out': 'Đăng xuất',
        'welcome-page': 'Chào mừng',
        'come-page': ' đến với trang HomePage',
        // validator registor
        RequireRegister: MessageConst.vi.Register.RequireRegister,
        EmailIsNotValid: MessageConst.vi.Register.EmailIsNotValid,
        userNameLengthBetweenFrom2To128Characters: MessageConst.vi.Register.userNameLengthBetweenFrom2To128Characters,
        PasswordLengthMustAtLeast6Characters: MessageConst.vi.Register.PasswordLengthMustAtLeast6Characters,
        Successfully: MessageConst.vi.Register.Successfully,
        ComfirmToFirstPassword: MessageConst.vi.Register.ComfirmToFirstPassword,
        AgeMinMax: MessageConst.vi.Register.AgeMinMax,
        AgeIntegerNumber: MessageConst.vi.Register.AgeIntegerNumber,
        EmailExists: MessageConst.vi.Register.EmailExists,
        SuccessRegister: MessageConst.vi.Register.SuccessRegister,

        // validator login
        RequireEmailAndPassword: MessageConst.vi.Login.RequireEmailAndPassword,
        ErrorAccount: MessageConst.vi.Login.ErrorAccount,
        LoginSuccess: MessageConst.vi.Login.LoginSuccess,
        NotExistAccounts: MessageConst.vi.Login.NotExistAccounts
      }
    }
  },
  fallbackLng: 'vi',
  // have a common namespace used around the full app
  ns: ['translations'],
  defaultNS: 'translations',

  keySeparator: false, // we use content as keys

  interpolation: {
    escapeValue: false, // not needed for react!!
    formatSeparator: ','
  },

  react: {
    wait: true
  }
});

export default i18n
