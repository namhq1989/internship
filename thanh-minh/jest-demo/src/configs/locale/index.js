import KeysConst from './key'
import vi from './vn'
import en from './en'

export {
  KeysConst,
  vi,
  en
}
