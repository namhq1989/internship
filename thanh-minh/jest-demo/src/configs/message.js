export default {
  vi: {
    Login: {
      RequireEmailAndPassword: 'Vui lòng điền đầy đủ thông tin đăng nhập',
      EmailIsNotValid: 'Email không đúng định dạng',
      PasswordLengthMustAtLeast6Characters: 'Mật khẩu phải có ít nhất 6 ký tự',
      Successfully: 'Đăng nhập thành công',
      ErrorAccount: 'Tên Email hoặc mật khẩu không đúng!',
      LoginSuccess: 'Đăng nhập thành công',
      NotExistAccounts: 'Không tồn tại tài khoản đăng nhập'
    },
    Register: {
      RequireRegister: 'Vui lòng điền đầy đủ thông tin đăng ký',
      EmailIsNotValid: 'Email không đúng định dạng',
      userNameLengthBetweenFrom2To128Characters: 'UserName phải có từ 2 -128 ký tự',
      PasswordLengthMustAtLeast6Characters: 'Mật khẩu phải có ít nhất 6 ký tự',
      Successfully: 'Đăng ký thành công',
      ComfirmToFirstPassword: ' Xác nhận mật khẩu không khớp',
      AgeMinMax: 'Độ tuổi phải nằm trong khoảng từ 1 đến 100',
      AgeIntegerNumber: 'Vui lòng nhập  tuổi  là một số nguyên',
      EmailExists: 'Email đã tồn tại. Vui lòng nhập email khác',
      SuccessRegister: 'Đăng ký thành công'
    }
  },
  en: {
    Login: {
      RequireEmailAndPassword: 'Please fill out the login form',
      EmailIsNotValid: 'The input is not valid E-mail!',
      PasswordLengthMustAtLeast6Characters: 'Passwords must be at least 6 characters',
      Successfully: 'Logged in successfully',
      ErrorAccount: 'Bad email address or password!',
      LoginSuccess: 'Logged in successfully',
      NotExistAccounts: 'No login account exists'
    },
    Register: {
      RequireRegister: 'Please fill out the registration form',
      EmailIsNotValid: 'The input is not valid E-mail!',
      userNameLengthBetweenFrom2To128Characters: 'UserName must be between 2 -128 characters',
      PasswordLengthMustAtLeast6Characters: 'Passwords must be at least 6 characters',
      Successfully: 'Sign Up Success',
      ComfirmToFirstPassword: 'Confirm password mismatch',
      AgeMinMax: 'Age must be in between 1 to 100',
      AgeIntegerNumber: 'Please enter age as an integer',
      EmailExists: 'Email already exists. Please enter a different email.',
      SuccessRegister: 'Sign Up Success'
    }
  }

}
