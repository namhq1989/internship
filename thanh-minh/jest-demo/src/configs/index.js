import ImageConst from './image'
import AppConst from './app'
import i18n from './i18n'
import AccountKitConst from './appId'

export {
  ImageConst,
  AppConst,
  i18n,
  AccountKitConst
}
