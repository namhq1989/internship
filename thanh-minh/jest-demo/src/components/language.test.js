import React from 'react'
import { shallow } from 'enzyme'
import { Button } from 'antd'
import ReactCountryFlag from 'react-country-flag'
import LanguageChange from './language'
import i18n from '../configs/i18n'
import { AppConst } from '../configs'


const changeLanguage = (lng) => {
  i18n.changeLanguage(lng)
}
const t = jest.fn()
test('Component LanguageChange render correctly', () => {
  const wrapper = shallow(
    <LanguageChange t={t} changeLanguage={changeLanguage} language="en" code="US" />
  )
  expect(wrapper).toMatchSnapshot()
})
test('Change language when click button flag', () => {
  const wrapper = shallow(
    <LanguageChange t={t} changeLanguage={changeLanguage} language="en" code="US" />
  )
  wrapper.simulate('click')
  expect(i18n.language).toEqual('vi')
})

test('should render 2 button language', () => {
  const wrapper = shallow(
    <LanguageChange t={t} changeLanguage={changeLanguage} language="en" code="US" />
  )
  expect(wrapper.find(Button)).toHaveLength(AppConst.language.length)
  expect(wrapper.find(ReactCountryFlag)).toHaveLength(AppConst.language.length)
})
