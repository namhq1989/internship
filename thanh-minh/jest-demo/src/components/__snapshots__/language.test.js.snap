// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component LanguageChange render correctly 1`] = `
<div
  className="country-flag"
>
  <span>
    :  
  </span>
  <Button
    className="margin-right"
    ghost={false}
    key="1"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
  >
    <ReactCountryFlag
      code="VN"
      svg={true}
    />
  </Button>
  <Button
    className="margin-right"
    ghost={false}
    key="2"
    loading={false}
    onClick={[Function]}
    prefixCls="ant-btn"
  >
    <ReactCountryFlag
      code="US"
      svg={true}
    />
  </Button>
</div>
`;
